### DISCUSSION TITLE
How to make the pore body as a node?

### DISCUSSION THREAD

**Discussion by morendong** [PMEAL/OpenPNM#2816] [Q&A] [answered]
Hi, everyone,
The pore network consists of pores body and throats, and the pore body's radius is larger than than throat's.
How to make the throat occupy all of the pore's volume, or make the pore body to be a site, that does not occupy volume?
Thanks!

**Comment by jgostick** [comment]
You can adjust the geometry pore-scale models to accomplish anything you need.  [This example](https://openpnm.org/examples/tutorials/05_defining_geometry.html) goes into detail about how to change or alter these models.  You probably should *not* set the pore diameter to 0 since this will affect the transport conductance models.  In fact, depending what you're trying to do you might want to write your own conductance model that assume the throat is 100% of the transport length and just ignores anything to do with the pores.

**Reply by morendong** [reply]
Thanks for your kind reply. As my understanding that your  meaning is that the ignoring the pore, just adjust the throat size?
If do as above, the porosity would decrease significantly.
In addition, the pore network was generated from Porespy, as I adjusted the size of pore and throat, which warning that the pore diameter is smaller than throat diameter. And the wide distribution of pore diameter can not generated.
Can you give me some advice?

**Reply by jgostick** [reply]
Well my main advice is that ignoring the pores is a bad idea, but I'm not sure what you're trying to do so I didn't say this earlier.  I can probably give you better advice if you tell me what you're trying to do, instead of asking me what you *think* you need to do.

**Reply by morendong** [reply]
I am trying to generate pore networks representing different soil structure and simulate invasion process.
The focus is the distribution of throat  diameter, mainly in the wide range and narrow range.
Now the wide range distribution of throat diameter is hard to generate.
In addition, the porosity of pore network is also needed to meet the real soil porosity.
Thanks in advance.

**Reply by jgostick** [reply, marked as answer]
If you think of the porous material as an image, the only way to have a wide range of sizes is to have a very, very large image so that you can resolve the small pores and large pores at the same time. We have basic example on generating [multiscale images here](https://porespy.org/examples/generators/tutorials/Creating_Multiscale_Images.html) to give you an idea.  

To accomplish this with a pore network you would also need a 'multiscale network', similar to the one we used in [this paper](https://doi.org/10.1016/j.cej.2017.07.139). So if you're trying to use a cubic pore network and assigning throat sizes, it will be impossible to accomplish 'multiscale' and match porosity and have consistent geometry (i.e. throats smaller than pores).  

Basically, you're pushing the envelop of what is 'normal' pore network modeling when you start to do multiscale, so you need to get creative.  I would suggest starting with a Delaunay and/or Voronoi network, and distributing the base points in a hierarchical way.
